### DISCUSSION TITLE
SIWE Next Auth Integration Fails Signature Verification After Connecting

### DISCUSSION THREAD

**Discussion by joeblau** [WalletConnect/walletconnect-monorepo#4270] [Developer Support] [answered]
https://docs.walletconnect.com/web3modal/nextjs/siwe/next-auth
 
When implementing this code above, after successfully connecting my wallet, the signature check always fails. On the backend api, it works properly with the provider singing in. Even after signature declined, the app still signs in.

What logic controls the signature?

<img width="332" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/1218847/935ae286-d10a-41d5-a215-60618e4b1f7c">

**Comment by glitch-txs** [comment]
hi, do you see any error on console? Can you share a minimal reproducible example?

**Comment by joeblau** [comment]
Console has no errors other than this one which seems to be a bug in the wallet connect API where the documentation says you can use `localhost` but you can't — but that's another issue 🫡

<img width="1296" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/1218847/b98f7246-985a-437b-a5a0-52f85f31b1c6">

It takes so much scaffolding to build up a project... Give me about 30 mins and I'll build a test project.

**Comment by joeblau** [comment, marked as answer]
okay @glitch-txs , I built a sample project. You would need to add a `NEXT_PUBLIC_PROJECT_ID=` to `.env` and then everything should work. 

https://github.com/joeblau/web3modal-demo

I've got good news and I've got bad news...

- Good news is that everything seems to work perfectly from the web3 modal.
-  Bad news is that I've done something to break the signature verification and I need to figure out where the problem is 🫡

**Comment by joeblau** [comment]
Ahhhh I found the bug [In this `signIn` closure ](https://github.com/joeblau/web3modal-demo/blob/main/src/contexts/index.tsx#L48)

```javascript
     const success = await signIn("credentials", {
        message,
        redirect: false,
        signature,
        callbackUrl: "/protected",
      });
```

if redirect is set to `true`, the UI seems to check the signature a second time which probably fails because the `nonce` is already used. If redirect is set to `false`, the screen dismisses properly.

**Reply by mcunni117** [reply]
I had the same issue and I can confirm that changing redirect to false removed the error message on a successful sign-in.  Do you know how/if it is possible to utilize the redirect functionality without the error or is it a bug in appkit and i'll need to use a workaround?
